src/main/java/org/apache/sysml/utils/Statistics.java

```diff
@@ -69,6 +72,16 @@
 	private static final LongAdder numExecutedSPInst = new LongAdder();
 	private static final LongAdder numCompiledSPInst = new LongAdder();
 
+	// number and size of pinned objects in scope
+	private static final DoubleAdder sizeofPinnedObjects = new DoubleAdder();
+	private static long maxNumPinnedObjects = 0;
+	private static double maxSizeofPinnedObjects = 0;
+
+	// Maps to keep track of CP memory objects for JMLC (e.g. in memory matrices and frames)
+	private static final ConcurrentHashMap<String,Double> _cpMemObjs = new ConcurrentHashMap<>();
+	private static final ConcurrentHashMap<Integer,Double> _currCPMemObjs = new ConcurrentHashMap<>();
+	private static final ConcurrentHashMap<Integer,SoftReference<CacheBlock>> _liveObjects = new ConcurrentHashMap<>();
```

Comment:
Can you add comment here about underutlization if this was handled via rmvar mechanism ?

Reply:
Done!